WcaOnRails/app/controllers/api/v0/roles_controller.rb

```diff
@@ -107,21 +118,67 @@ def index_for_group_type
     group_ids = UserGroup.where(group_type: group_type).pluck(:id)
     roles = Role.where(group_id: group_ids).to_a # to_a is for the same reason as in index_for_user.
 
-    # Temporary hack to support the old delegate structure, will be removed once all roles are
+    # Temporary hack to support the old system roles, will be removed once all roles are
     # migrated to the new system.
     if group_type == UserGroup.group_types[:delegate_regions]
       roles.concat(User.where.not(delegate_status: nil).map(&:delegate_role))
+    elsif group_type == UserGroup.group_types[:councils]
+      Team.all_councils.each do |council|
+        leader = TeamMember.find_by(team_id: council.id, team_leader: true, end_date: nil)
```

Comment:
This _should_ be eqauivalent to (please double-check) `council.leader`. Or something equivalent, maybe it's called `council.current_leader` or similar. In any case, I'd be very surprised if `Team` model objects didn't have a convenience method for retrieving the leader.